avaframe/in3Utils/geoTrans.py

```diff
@@ -309,14 +320,68 @@ def remeshDEM(cfg, dem):
         headerRemeshed['yllcenter'] = headerDEM['yllcenter']
         headerRemeshed['noDataValue'] = headerDEM['noDataValue']
 
-        dem['header'] = headerRemeshed
+        # write new DEM dictionary
+        remeshedDEM = {'header': headerRemeshed}
         xNewGrid, yNewGrid = np.meshgrid(xNew, yNew)
         zNew = sp.interpolate.griddata((xGrid, yGrid), z, (xNewGrid, yNewGrid), method='cubic',
                                        fill_value=headerDEM['noDataValue'])
         log.info('Remeshed data extent difference x: %f and y %f' % (diffExtentX, diffExtentY))
-        dem['rasterData'] = zNew
+        remeshedDEM['rasterData'] = zNew
+
+        # save remeshed DEM
+        pathToDem = pathlib.Path(cfg['avalancheDir'], 'Inputs', 'DEMremeshed')
+        fU.makeADir(pathToDem)
+        outFile = pathToDem / ('remeshedDEM%.2f.asc' % remeshedDEM['header']['cellsize'])
+        IOf.writeResultToAsc(remeshedDEM['header'], remeshedDEM['rasterData'], outFile, flip=True)
+        log.info('Saved remeshed DEM to %s' % outFile)
+
+    return remeshedDEM
+
+
+def searchRemeshedDEM(cfg, dem):
+    """ search if remeshed DEM already available and check if cellSize and origin are correct
+
+        Parameters
+        -----------
+        cfg: configparser object
+            configuration settings: avaDir, meshCellSize, meshCellSizeThreshold
+        dem: dict
+            dictionary of DEM in Inputs
+
+        Returns
+        --------
+        remshedDEM: dict
+            dictionary of remeshed DEM if not found empty dict
+        DEMFound: bool
+            flag if dem is found
+    """
+
+    # load input parameters
+    cellSize = cfg.getfloat('meshCellSize')
+    cellSizeThreshold = cfg.getfloat('meshCellSizeThreshold')
+
+    # path to remeshed DEM folder
+    pathToDem = pathlib.Path(cfg['avalancheDir'], 'Inputs', 'DEMremeshed')
+    DEMFound = False
+    remeshedDEM = {}
+    if pathToDem.is_dir():
+        # look for dems and check if cellSize within tolerance and origin matches
+        demFiles = list(pathToDem.glob('*.asc'))
```

Comment:
split this here and move to for to the same indent as the next for. Do a else return/etc on the if pathToDem